lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -81,12 +81,46 @@ openModuleFiles(StringRef DirName, StringRef ModuleFilename,
   return std::error_code();
 }
 
+static void addDiagnosticInfoForArchitectureMismatch(ASTContext &ctx,
+                             SourceLoc sourceLocation,
+                             llvm::SmallString<64> moduleName,
+                             llvm::SmallString<16> archName,
+                             llvm::SmallString<128> directoryPath) {
+
+  std::error_code errorCode;
+  llvm::Twine twineDirPath(directoryPath);
+  llvm::sys::fs::directory_iterator directoryIterator(twineDirPath,
+                                                      errorCode,
+                                                      true);
+  llvm::sys::fs::directory_iterator endIterator;
+
+  if(errorCode) {
+      return;
+    }
+
+  std::string foundArchs;
+  for (; directoryIterator != endIterator; directoryIterator.increment(errorCode)) {
+    auto entry = *directoryIterator;
+    llvm::StringRef filePath(entry.path());
+    llvm::StringRef extension = llvm::sys::path::extension(filePath);
```

Comment:
Nitpick: StringRef and a handful of other types from the LLVM namespace are so common that it's conventional not to use `llvm::` with them in Swift code. You can see the list of such types in the "swift/Basic/LLVM.h" header.

(You'll notice SmallString is in this list too, so clearly we're not perfectly consistent about it ourselves. But StringRef is super common.)